src/hist/plot.py

```diff
@@ -210,159 +223,334 @@ def plot2d_full(
+    if uncert_type == "poisson":
+        ratio_uncert = np.abs(poisson_interval(ratio, num / np.square(denom)) - ratio)
+        # TODO: Why does SciPy not work here?
+        # coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+        # alpha = 1 - coverage
+        # ratio_uncert = stats.poisson.interval(alpha, ratio)
+    elif uncert_type == "poisson-ratio":
+        # poisson ratio n/m is equivalent to binomial n/(n+m)
+        ratio_uncert = np.abs(clopper_pearson_interval(num, num + denom) - ratio)
+    return ratio_uncert
+
+
+def filter_kwargs(kwargs: Dict[str, Any]) -> Tuple[Dict[str, Any], ...]:
+    """
+    Filter kwargs into relevant subsets
+    """
 
     # error bar keyword arguments
     eb_kwargs = _filter_dict(kwargs, "eb_")
     eb_kwargs.setdefault("label", "Histogram Data")
-    eb_kwargs.setdefault("fmt", "o")
+    eb_kwargs.setdefault("marker", "o")
+    eb_kwargs.setdefault("linestyle", "none")
 
     # fit plot keyword arguments
-    label = "Fit"
-    if fit_fmt is not None:
-        for name, value, error in zip(parnames, popt, perr):
-            label += "\n  "
-            label += fit_fmt.format(name=name, value=value, error=error)
     fp_kwargs = _filter_dict(kwargs, "fp_")
-    fp_kwargs.setdefault("label", label)
+    fp_kwargs.setdefault("label", "Counts")
+
+    # bar plot keyword arguments
+    bar_kwargs = _filter_dict(kwargs, "bar_", ignore={"bar_width"})
 
     # uncertainty band keyword arguments
     ub_kwargs = _filter_dict(kwargs, "ub_")
     ub_kwargs.setdefault("label", "Uncertainty")
-    ub_kwargs.setdefault("alpha", 0.5)
 
-    # bar plot keyword arguments
-    bar_kwargs = _filter_dict(kwargs, "bar_", ignore={"bar_width"})
+    # ratio plot keyword arguments
+    rp_kwargs = _filter_dict(kwargs, "rp_")
+    rp_kwargs.setdefault("uncert_type", "poisson")
+    rp_kwargs.setdefault("legend_loc", "best")
 
     # patch plot keyword arguments
-    pp_kwargs = _filter_dict(kwargs, "pp_", ignore={"pp_num"})
-    pp_num = kwargs.pop("pp_num", 5)
+    pp_kwargs = _filter_dict(kwargs, "pp_")
 
     # Judge whether some arguments are left
     if kwargs:
         raise ValueError(f"{set(kwargs)}' not needed")
 
-    # Main: plot the pulls using Matplotlib errorbar and plot methods
-    main_ax.errorbar(self.axes.centers[0], ydata, yerr, **eb_kwargs)
+    return eb_kwargs, fp_kwargs, bar_kwargs, ub_kwargs, rp_kwargs, pp_kwargs
+
+
+def _construct_gaussian_callable(
+    _hist: hist.BaseHist,
+) -> Callable[[np.ndarray, float, float, float], np.ndarray]:
+    x_values = _hist.axes[0].centers
+    hist_values = _hist.values()
+
+    # gaussian with reasonable initial guesses for parameters
+    constant = float(hist_values.max())
+    mean = (hist_values * x_values).sum() / hist_values.sum()
+    sigma = (hist_values * np.square(x_values - mean)).sum() / hist_values.sum()
+
+    # TODO: Figure out why type must be ignored to pass mypy
+    # error: Returning Any from function declared to return "ndarray"
```

Comment:
Same issue, NumPy returns Any from ufuncts. Eventually this will be improved, I hope, but that's what we have for now.